Issue/accessibility

dmols opened the pull request:
Updated Simple Survey to have better accessibility. Some key things to note that were implemented: 

- Screen reader used to ignore the questions listed and now it reads them just fine.
- Added the radio buttons to the tab sequence; previously they were skipped. Also removed the radio groups from the tab sequence because there's no good reason to have it in the tab order. 
- Background changes to light green on focus, to increase readability and allow the user to easily keep track of where they are in the tab sequence and where the screen reader's focus is on.

_Note_: I overlooked disabling a syntax formatter I have enabled, and so a lot of the changes in `player.html` are mostly indentation fixes. Apologies for the incovenience when reviewing.

FrenjaminBanklin reviewed (changes requested):
I'd say it's worth going back into the player.html file and changing indentation from spaces back to tabs, if only to make the diff readable.

clpetersonucf reviewed (changes requested):
Great so far, here's some issues I ran into:

- As requested by Brandon, we prefer to use tabs instead of spaces for indentation. Converting the entire file from one form of indentation to another makes the diff unreadable.
- Left/right arrow keys don't yet work to switch focus between questions, at least in Firefox.
- Radio buttons cannot be selected via `enter` or `space`.
- The list box controls work, but the "Accept Order" button is being interpreted as one of the list items. It's not a huge deal, but it means that a list of 6 items will be read as "X of 7" instead of "X of 6", which is super confusing.
- The frame of the widget is still being read as "Privilege Walk", which is the widget that Survey pulled from in development. It should indicate "Simple Survey Materia Widget" instead and ideally provides the name of the widget instance, ex: "Severe Weather Survey: Simple Survey Materia Widget".

dmols commented:
Great suggestions, thank you guys. I've been looking into the changes needed and I'll keep you updated.

clpetersonucf reviewed (approved):
Works intuitively, and screenreader navigation is solid. The diff looks good. I don't have any additional feedback.

FrenjaminBanklin reviewed (approved):
If possible it might be nice to have a little blurb in an aria-label somewhere about the left/right arrow keys moving quickly between questions, but otherwise this looks good and functions well.

dmols commented:
Updated my branch with One-At-A-Time changes, and increased accessibility for that format of the survey. I also added a tooltip to the toolbar (for instructions on question navigation). Made it sort of 'punny' so feel free to let me know if you want the icon and text to be more serious, though I think the user would understand regardless. Thanks!

clpetersonucf commented on src/creator.coffee:
```diff
@@ -94,6 +96,7 @@ SurveyWidget.controller 'SurveyWidgetController', [ '$scope','$mdToast','$mdDial
 		$scope.$apply ->
 			$scope.title = title
 			$scope.groups = qset.options.groups
+			$scope.OneQuestionAtATime = qset.options.OneQuestionAtATime
```
We need to ensure the creator and player are backwards compatible with previously created qsets which will not have this property. There should be a check here to ensure that if the `OneQuestionAtATime` value is not present, it should be handled appropriately by being set to false.

clpetersonucf reviewed (changes requested):
Additional feedback with the addition of the one-at-a-time branch:

- The player should indicate the current question index and total number of questions visually when in one-at-a-time mode.
- Questions in one-at-a-time mode do not read their question content and index position aloud, as they do in the standard mode.
- There should be feedback when paginating between questions, via an `aria-live` region. It should indicate whether we've moved forward or backwards and the new question index.
- After paginating, we should programmatically focus the now-current question.

clpetersonucf reviewed (changes requested):
I think it's almost ready. A few remaining things:

- Selecting an answer choice with the keyboard does not appear to trigger completion tracking the same way selecting with a mouse does. If I use the keyboard to answer every question, the submit button is not highlighted and trying to submit the widget gives me the message that I need to complete all questions, despite having done so.
- I would prefer the question index and count, for one-at-a-time mode, to not be focusable, since it's not an input. That element is good for visual feedback but the question position text ("Question X of Y") should be prepended to the question text itself when read by the screen reader when the question is focused. Ex: "Question 1 of 2: What is the airspeed velocity of an unladen swallow?"
- In one-at-a-time mode, the question element appears to shift horizontally when paginating with the arrow keys. This may be a firefox thing?

clpetersonucf reviewed (approved):
Looks good to go. No additional issues to address.
